src/document.jl

```diff
@@ -113,24 +113,20 @@ function get_offset2(doc::Document, line::Integer, character::Integer)
     pos = line_offset
 
     while character > 0
-        try
-            if UInt32(text[pos]) >= 0x010000
-                character -= 2
-            else
-                character -= 1
-            end
-        catch err
-            if err isa BoundsError
-                throw(LSOffsetError("get_offset2 crashed at location 3. More diagnostics:\nline=$line\nline_offsets='$line_offsets'\ncharacter=$character\npos=$pos\nlastindex=$(lastindex(text))"))
-            else
-                rethrow(err)
-            end
+        if pos >= next_line_offset
+            pos = next_line_offset
+            break
         end
+
+        if UInt32(text[pos]) >= 0x010000
+            character -= 2
+        else
+            character -= 1
+        end
+
         pos = nextind(text, pos)
     end
 
-    ret = min(pos, next_line_offset)
```

Comment:
Note that this line actually was never effective because we returned `pos` rather than `ret` from the function, which was a bug in itself. The new code with the `if` statement in the loop should be equivalent to this, _and_ also run actually :)

Reply:
Is this actually something the linter could catch? Assigning something to a local variable that then gets never used seems like worthy a warning?

Reply:
Yeah it is, I'll make PR

Reply:
@ZacLN is that something StaticLint.jl could do?